modules/odc-deploying-java-applications.adoc

```diff
@@ -0,0 +1,22 @@
+// Module included in the following assemblies:
+//
+// applications/application_life_cycle_management/odc-deploying-java-applications.adoc
+
+[id="odc-deploying-java-applications_{context}"]
+= Upload JAR files for easy deployment of Java applications
+
+You can use the JAR files in the *Topology* view of the *Developer* perspective to deploy your Java applications.
+
+The following procedure explains the *Upload JAR file* option in the *Developer* perspective to deploy a Java application.
+
+.Procedure
+
+. In the *Topology* view, click the *Add to Project* menu item, which appears when you right-click on the *Topology* surface to see the *Upload JAR file* option. You can alternatively drag and drop the JAR file on the *Topology* surface.
+. Click on the *Upload JAR file* option to open the *Upload JAR file* form.
+. In the *Upload JAR file* form:
+* In the *JAR file* field, browse the JAR file on your local machine and upload the same or you can drag-and-drop it on the field.
+* In the *Application Name* field, specify a unique name for your application to label the resources.
+* In the *Name* field, specify a unique component name to name the associated resources.
+* In the *Resources* field, choose the resource type for your application.
+* In the *Advanced options*, click on *Create a Route to the Application* to configure a public URL for your deployed application.
+. Click *Create* to deploy the application. Once the JAR file upload is completed and deployment is added to topology, you can check the build status badge which reflects the status of the application.
```

Comment:
- if there are no errors while submitting the form then the user will see a toast notification on the top right saying that the JAR file is uploading and it might take a while. The toast notification also includes a link to view the build logs. 

- If the user attempts to close the browser then a web alert would appear asking the user if they really wanted to leave the page

Reply:
Fixed!